### PR TITLE
feat: migrate schema queries to typescript

### PR DESCRIPTION
**Pull request by lohart13**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

This PR migrate the `createSchema`, `dropSchema` and `dropAllSchemas` queries to typescript.

# Breaking Changes
`dropSchema` no longer cascades by default. You will need to manually specify `cascade: true` in the options for dialects that support it otherwise drop all objects in the schema before dropping the schema. See the `dropAllSchemas` method for an example.

### CODE REVIEW

File: packages/core/src/dialects/mariadb/index.ts [hunk 1]
```diff
@@ -61,6 +61,17 @@ export class MariaDbDialect extends AbstractDialect {
       removeColumn: {
         ifExists: true,
       },
+      createSchema: {
+        charset: true,
+        collate: true,
+        // TODO: uncomment when MariaDB 10.5 is oldest supported version
```
Comment (WikiRik) [inline packages/core/src/dialects/mariadb/index.ts:67]:
```suggestion
        // TODO [>=2024-06-19]: uncomment when MariaDB 10.5 is oldest supported version
```

File: packages/core/test/integration/query-interface/schemas.test.ts [hunk 2]
```diff
@@ -32,6 +31,68 @@ describe('QueryInterface#{create,drop,list}Schema', () => {
     expect(postCreationSchemas).to.include(testSchema, 'createSchema did not create testSchema');
   });
 
+  if (dialect.supports.createSchema.authorization) {
```
Comment (WikiRik) [inline packages/core/test/integration/query-interface/schemas.test.ts:34]: For these new tests; do we want to validate in some way that the options are actually used? Or is relying on the unit tests for this enough?

**Review by WikiRik — COMMENTED** [review]
Two small remarks/questions. I still need to look at the DB2 code since a bigger change is happening there. I might review other PRs of you before I get to this, so therefore already this preliminary review

File: packages/core/test/integration/query-interface/schemas.test.ts (hunk 2 above)
Comment (lohart13) [reply packages/core/test/integration/query-interface/schemas.test.ts:34]: Sure, I've updated the test now.
With db2 you cannot create users using SQL statements therefore I left it to use `CURRENT_USER` as this usually defaults to the logged in user.

Comment (WikiRik) [reply packages/core/test/integration/query-interface/schemas.test.ts:34]: Thanks for this! I was also talking about the other options, could you add additional checks for those as well?

Comment (lohart13) [reply packages/core/test/integration/query-interface/schemas.test.ts:34]: Sure, added now

**Review by WikiRik — APPROVED** [review]

**Comment by WikiRik** [comment]
For this and breaking changes in your other PRs; can you open a PR in https://github.com/sequelize/website to document them?

**Comment by lohart13** [comment]
Tagging PR to sequelize/website#501

**Comment by WikiRik** [comment]
@lohart13 I think this PR introduced some flakyness back in for DB2, seems that the simplification still had some side-effects. This error occured `SequelizeDatabaseError: [IBM][CLI Driver][DB2/LINUXX8664] SQL0911N  The current transaction has been rolled back because of a deadlock or timeout.  Reason code "2".  SQLSTATE=40001`. Check out the rest of the run [here](https://github.com/sequelize/sequelize/actions/runs/7717004571/job/21035555464?pr=16993).
Can you take a look at this?

**Comment by lohart13** [comment]
@WikiRik please see #17024 which should fix the deadlocks when dropping the test db's

**Pull request merged**